src/routes/bounce_rules_activity.jsx

```diff
@@ -0,0 +1,228 @@
+    this.updatePageIndex = this.updatePageIndex.bind(this);
+    this.handleActivityClicked = this.handleActivityClicked.bind(this);
+    this.handlePrevClicked = this.handlePrevClicked.bind(this);
+    this.handleNextClicked = this.handleNextClicked.bind(this);
+  }
+
+  async componentDidMount() {
+    const { data: activities } = await getActivityLog();
+    if (activities) {
+      this.setState({
+        isFetching: false,
+        activityLog: activities.reverse(),
+      });
+    }
+  }
+
+  logout() {
+    const { history } = this.props;
+    localStorage.clear();
+    history.push("/");
+  }
+
+  handleModalClose(e) {
+    const { id } = e.currentTarget;
+    this.setState({
+      [id]: false,
+      isInvalidInput: false,
+      selectedRule: {},
+    });
+  }
+
+  updateSearchToken(e) {
+    this.setState({
+      searchToken: e.target.value.toLowerCase(),
+    });
+  }
+
+  updateSearchCategory(e) {
+    this.setState({
+      searchCategory: e.value.toLowerCase(),
+    });
+  }
+
+  filterRules(rules) {
+    const { searchToken } = this.state;
+    return rules.filter(
+      rule =>
+        rule.bounce_action.toLowerCase().includes(searchToken) ||
+        rule.description.toLowerCase().includes(searchToken)
+    );
+  }
+
+  paginate(activityLog) {
+    const { currentActivityPageIndex, rulesToShow } = this.state;
+    const ruleStartIndex = (currentActivityPageIndex - 1) * rulesToShow;
+    const ruleEndIndex =
+      (currentActivityPageIndex - 1 * currentActivityPageIndex + rulesToShow) *
+      currentActivityPageIndex;
+    return activityLog.slice(ruleStartIndex, ruleEndIndex);
+  }
+
+  updatePageIndex(e) {
+    const newIndex = parseInt(e.currentTarget.getAttribute("value"), 10);
+    this.setState(prevState => {
+      const isPageIndexUpdated =
+        prevState.currentActivityPageIndex !== newIndex;
+      return {
+        currentActivityPageIndex: isPageIndexUpdated
+          ? newIndex
+          : prevState.currentActivityPageIndex,
+      };
+    });
+  }
+
+  handlePrevClicked() {
+    this.setState(prevState => ({
+      currentActivityPageIndex: prevState.currentActivityPageIndex - 1,
+    }));
+  }
+
+  handleNextClicked() {
+    this.setState(prevState => ({
+      currentActivityPageIndex: prevState.currentActivityPageIndex + 1,
+    }));
+  }
+
+  isDuplicate(searchCategory, searchToken) {
+    let found = false;
+    const { filterOptions } = this.state;
+    filterOptions.filter(filter => {
+      if (
+        filter.searchCategory === searchCategory &&
+        filter.searchToken === searchToken
+      ) {
+        found = true;
+      }
+      return false;
+    });
+    return found;
+  }
+
+  addFilter() {
+    const { searchCategory, searchToken } = this.state;
+    if (!searchCategory || !searchToken) {
+      this.setState({
+        invalidFilter: true,
+      });
+      return;
+    }
+    if (!this.isDuplicate(searchCategory, searchToken)) {
```

Comment:
Let's put the isDuplicate condition first. It's easier to read positive ones first and then in the else do the negative so it would read like ```this.isDuplicate() { ... } else { ...negative part }``` for less flipping